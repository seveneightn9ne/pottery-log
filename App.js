// @flow
import React from 'react';
import AppContainer from './src/AppContainer.js';
import {AsyncStorage} from 'react-native';
import {Font} from 'expo';
import Sentry from 'sentry-expo';
// import { SentrySeverity, SentryLog } from 'react-native-sentry';
Sentry.config('https://bad3ce09a6404191b1a0b181bdb36263@sentry.io/1341402').install();

export default class App extends React.Component<any, {fontLoaded: boolean}> {
  constructor(props) {
    super(props);
    this.state = {fontLoaded: false};
  }

  async componentDidMount(): Promise<void> {
    await Font.loadAsync({
      'material-icons': require('./assets/MaterialIcons-Regular.ttf'),
    });
    this.setState({fontLoaded: true});
  }
  render() {
    return <AppContainer fontLoaded={this.state.fontLoaded} />
  }
}
