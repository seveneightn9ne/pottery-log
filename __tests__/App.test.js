import React from "react";
import App from "../App";
import { shallow, mount } from "enzyme";

describe("App", () => {
  it("matches the snapshot after font loaded", async () => {
    const wrapper = await shallow(<App />);
    await wrapper.instance().componentDidMount();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state("fontLoaded")).toEqual(true);
  });

  it("can render", () => {
    mount(<App />);
  });
});
