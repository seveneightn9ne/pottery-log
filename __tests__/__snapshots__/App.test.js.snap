// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot after font loaded 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(AppView)
    fontLoaded={true}
  />
</Provider>
`;
